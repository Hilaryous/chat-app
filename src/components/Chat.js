// @flow
import React, { PureComponent } from 'react'
import styled from 'react-emotion'
import type ChatType from '../types/Chat'
import Button from './Button'

type Props = {
  chat: ChatType,
  selectChat: () => void,
}

class Chat extends PureComponent<Props> {
  handleSelectChat = () => {
    const { selectChat, chat: { id } } = this.props
    selectChat(id)
  }

  render() {
    const { chat } = this.props
    return (
      <Container id="chat">
        <User>
          {chat.userEmail}
        </User>
        <Button type="button" onClick={this.handleSelectChat}>
        Go to Chat
        </Button>
      </Container>
    )
  }
}

export default Chat

// --------------------------

const Container = styled('div')({
  border: '1px solid black',
  fontFamily: 'sans-serif',
  padding: '16px',
})

const User = styled('div')({
  fontWeight: 'bold',
})
