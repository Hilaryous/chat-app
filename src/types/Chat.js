// @flow
export type ChatType = {
  id: string,
  toUser: string,
  fromUser: string,
}
