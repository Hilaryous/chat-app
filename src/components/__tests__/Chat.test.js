import React from 'react'
import renderer from 'react-test-renderer'
import Chat from '../Chat'

describe('Chat', () => {
  it('shows the correct snapshot tree for the Chat component', () => {
    const tree = renderer.create(
      <Chat chat={{ userEmai: 'foo@test.com' }} selectChat={() => {}} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
