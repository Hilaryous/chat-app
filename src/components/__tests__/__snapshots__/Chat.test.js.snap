// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat shows the correct snapshot tree for the Chat component 1`] = `
<div
  className="css-gnrrs"
>
  <div
    className="css-in3yi3"
  />
  <button
    className="css-bjolns"
    onClick={[Function]}
    type="button"
  >
    Go to Chat
  </button>
</div>
`;
